pkg/agent/manager/cache/cache.go

```diff
@@ -445,6 +536,109 @@ func (c *Cache) GetStaleEntries() []*StaleEntry {
 	return staleEntries
 }
 
+// SyncSVIDsWithSubscribers will sync svid cache:
+// entries with active subscribers which are not cached will be put in staleEntries map
+// records which are not cached for remainder of max cache size will also be put in staleEntries map
+func (c *Cache) SyncSVIDsWithSubscribers() {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	c.syncSVIDs()
+}
+
+// Notify subscribers of selector set only if all SVIDs for corresponding selector set are cached
+// It returns whether all SVIDs are cached or not.
+// This method should be retried with backoff to avoid lock contention.
+func (c *Cache) Notify(selectors []*common.Selector) bool {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	if len(c.missingSVIDRecords(selectors)) == 0 {
+		set, setFree := allocSelectorSet(selectors...)
+		defer setFree()
+		c.notifyBySelectorSet(set)
+		return true
+	}
+	return false
+}
+
+func (c *Cache) missingSVIDRecords(selectors []*common.Selector) []*StaleEntry {
+	set, setFree := allocSelectorSet(selectors...)
+	defer setFree()
+
+	records, recordsDone := c.getRecordsForSelectors(set)
+	defer recordsDone()
+
+	if len(records) == 0 {
+		return nil
+	}
+	out := make([]*StaleEntry, 0, len(records))
+	for record := range records {
+		if _, ok := c.svids[record.entry.EntryId]; !ok {
+			out = append(out, &StaleEntry{
+				Entry: record.entry,
+			})
+		}
+	}
+	return out
+}
+
+func (c *Cache) updateLastAccessTimestamp(selectors []*common.Selector) {
+	set, setFree := allocSelectorSet(selectors...)
+	defer setFree()
+
+	records, recordsDone := c.getRecordsForSelectors(set)
+	defer recordsDone()
+
+	for record := range records {
+		// Set lastAccessTimestamp so that svid LRU cache can be cleaned based on this timestamp
+		record.lastAccessTimestamp = c.clk.Now().UnixMilli()
+	}
+}
+
+// entries with active subscribers which are not cached will be put in staleEntries map
+// records which are not cached for remainder of max cache size will also be put in staleEntries map
+func (c *Cache) syncSVIDs() (map[string]struct{}, []record) {
+	activeSubs := make(map[string]struct{})
+	lastAccessTimestamps := make([]record, len(c.records))
+
+	i := 0
+	// iterate over all selectors from cached entries and obtain:
+	// 1. entries that have active subscribers
+	//   1.1 if those entries don't have corresponding SVID cached then put them in staleEntries
+	//       so that SVID will be cached in next sync
+	// 2. get lastAccessTimestamp of each entry
+	for id, record := range c.records {
+		for _, sel := range record.entry.Selectors {
+			if index, ok := c.selectors[makeSelector(sel)]; ok && index != nil {
+				if len(index.subs) > 0 {
+					if _, ok := c.svids[record.entry.EntryId]; !ok {
```

Comment:
```suggestion
					if _, svidCached := c.svids[record.entry.EntryId]; !svidCached {
```
This improves the readability as well as disambiguates the value of `ok` across the levels of nested `if`s, since this statement as written overrides the previous value of `ok`.